internal/gatewayapi/translator.go

```diff
@@ -459,16 +466,18 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR *ir.Xds,
 					proto = ir.HTTPSProtocolType
 				}
 				infraPort := ir.ListenerPort{
-					Name:          irInfraPortName(listener),
+					Name:          string(listener.Name),
 					Protocol:      proto,
 					ServicePort:   servicePort,
 					ContainerPort: containerPort,
```

Comment:
you'll also need to create a unique (servicePort, containerPort) tuple to ensure traffic doesnt get multiplexed in the envoy listener